frontend/packages/dev-console/src/components/topology2/nodeContextMenu.tsx

```diff
@@ -15,18 +22,31 @@ const onKebabOptionClick = (option: KebabOption) => {
   }
 };
 
-const createMenuItem = (action: KebabOption) => (
-  <ContextMenuItem
-    className="odc2-topology-context-menu__kebab-wrapper"
-    key={action.label}
-    onClick={() => onKebabOptionClick(action)}
-  >
-    <KebabItem option={action} onClick={null} />
-  </ContextMenuItem>
-);
+type kebabContextMenuItemProps = {
+  option: KebabOption;
+};
+
+const KebabContextMenuItemConnected = (
+  props: kebabContextMenuItemProps & { impersonate: string },
+) => {
+  const { option, impersonate } = props;
+  const isAllowed = useAccessReview(option.accessReview, impersonate);
+  return (
+    <ContextMenuItem
+      className="odc2-topology-context-menu__kebab-wrapper"
+      key={option.label}
+      onClick={() => onKebabOptionClick(option)}
+      isDisabled={option.isDisabled || !isAllowed}
+    >
+      <KebabItem option={option} onClick={null} />
+    </ContextMenuItem>
+  );
+};
```

Comment:
What if you use the PF4 `DropdownContext` like this:
```
const createMenuItem = (option: KebabOption) => (
  <DropdownContext.Consumer key={option.label}>
    {({ onSelect }) => (
      <KebabItem
        option={option}
        onClick={() => {
          onKebabOptionClick(option);
          onSelect();
        }}
      />
    )}
  </DropdownContext.Consumer>
);
```

Reply:
The issue is that the KebabItem changes once the accessReview is retrieved. The context menu is getting a ref from the KebabItem and saving it away. Since the KebabItem changes the saved ref is no longe valid and setting focus and key handlers to it is useless.

Using the wrapper ContextMenuItem causes the ref to remain constant and focus and key handlers then work correctly.

Reply:
The issue is that the KebabItem changes once the accessReview is retrieved. The context menu is getting a ref from the KebabItem and saving it away. Since the KebabItem changes the saved ref is no longe valid and setting focus and key handlers to it is useless.

Using the wrapper ContextMenuItem causes the ref to remain constant and focus and key handlers then work correctly.

Reply:
Updated such that the kebab item does not change once the access review is retrieved simplifying and unnesting!